deck.rb

```diff
@@ -7,11 +7,44 @@ class Deck
 
   SUITS = %w(hearts diamonds spades clubs)
 
+  SUITS.each do |suit|
+    define_method("#{suit}".to_sym) do
+      cards.select { |card| card.suit == suit }
+    end
+  end
+
   def initialize
     @cards = create_cards
-    shuffle
   end
 
+  def shuffle
+    cards.shuffle!
+  end
+
+  def deal_card
+    cards.pop
+  end
+
+  # All of the below methods are now created by the block on line 10
+
+  # def spades
+  #   cards.select { |card| card.suit == "spades" }
+  # end
+
+  # def hearts
+  #   cards.select { |card| card.suit == "hearts" }
+  # end
+
+  # def diamonds
+  #   cards.select { |card| card.suit == "diamonds" }
+  # end
+
+  # def clubs
+  #   cards.select { |card| card.suit == "clubs" }
+  # end
+
```

Comment:
@CorainChicago -- check it out! Now we have suit methods that return all the cards of a specific suit. The non dry implementation from lines 28-44 can be replaced with some metaprogramming on line 10